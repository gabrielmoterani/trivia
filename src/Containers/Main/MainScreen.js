import React from 'react';
import { connect } from 'react-redux';
import { Layout, Text, Spinner, Button } from 'react-native-ui-kitten';
import BaseLayout from '../../Components/BaseLayout';
import GameActions from '../../Stores/Game/Actions';
import i18n from '../../Locales/i18n';
import Style from './MainScreenStyle';

export function MainScreen(props) {
  return (
    <BaseLayout>
      { props.loading && (<Spinner size="giant" />) }
      { !props.loading && (
        <Layout style={Style.intro}>
          <Text category="s1">{i18n.t('main.intro')}</Text>
          <Text category="s2">{i18n.t('main.teaser')}</Text>
          <Button size="large" style={Style.beginBtn} onPress={props.newGame}>
            {i18n.t('main.start')}
          </Button>
        </Layout>
      ) }
    </BaseLayout>
  );
}

const mapStateToProps = (state) => ({
  loading: state.game.get('loading'),
});

const mapDispatchToProps = (dispatch) => ({
  newGame: () => dispatch(GameActions.newGame()),
});

export default connect(mapStateToProps, mapDispatchToProps)(MainScreen);
