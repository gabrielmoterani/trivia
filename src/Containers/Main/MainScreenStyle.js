import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    flex: 0.2,
    justifyContent: 'flex-start',
    marginTop: 50,
  },
  main: {
    flex: 0.8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  footer: {
    flex: 0.2,
    justifyContent: 'flex-end',
    marginBottom: 50,
  },
});
