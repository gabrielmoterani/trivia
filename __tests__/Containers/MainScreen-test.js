import React from 'react';
import { shallow } from 'enzyme';
import { MainScreen } from '../../src/Containers/Main/MainScreen';

const initialProps = {
  loading: false,
  newGame: jest.fn(),
};

const setup = (props = initialProps) => shallow(<MainScreen {...props} />);

describe('Test MainScreen container', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show new game button while not loading', () => {
    const newGameBtn = wrapper.find('ButtonComponent');
    expect(newGameBtn).toExist();
  });

  it('should trigger newGame whe button is pressed', () => {
    const newGameBtn = wrapper.find('ButtonComponent');
    newGameBtn.first().props().onPress();

    expect(initialProps.newGame).toHaveBeenCalled();
  });

  it('should hide new game button while loading', () => {
    wrapper.setProps({ loading: true });

    const newGameBtn = wrapper.find('ButtonComponent');

    expect(newGameBtn).not.toExist();
  });
});
