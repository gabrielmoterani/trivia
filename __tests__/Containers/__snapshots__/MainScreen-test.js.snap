// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MainScreen container should render properly 1`] = `
<BaseLayout>
  <LayoutComponent
    style={
      Object {
        "paddingHorizontal": 25,
        "textAlign": "center",
      }
    }
  >
    <TextComponent
      category="s1"
    >
      trivia is a simple "boolean" trivia game
    </TextComponent>
    <TextComponent
      category="s2"
    >
      are you good enough to score 100%?
    </TextComponent>
    <ButtonComponent
      onPress={[MockFunction]}
      size="large"
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      Start!
    </ButtonComponent>
  </LayoutComponent>
</BaseLayout>
`;
